maintenance_request_time_performance/models/maintenance.py

```diff
@@ -0,0 +1,149 @@
+from odoo import api, models, fields, _
+from datetime import timedelta
+from odoo.exceptions import ValidationError
+
+
+class MaintenanceStage(models.Model):
+    _inherit = 'maintenance.stage'
+
+    in_progress = fields.Boolean(
+        'In Progress',
+        default=False)
+
+    @api.constrains('in_progress', 'done')
+    def check_maintenance_stage(self):
+        if self.in_progress and self.done:
+            raise ValidationError(
+                _("It can´t be in progress and done at the same time"))
+
+
+class MaintenanceRequest(models.Model):
+    _inherit = 'maintenance.request'
+
+    start_request_datetime = fields.Datetime(
+        string='Start Date and time of maintenance request completion')
+    start_request_date = fields.Date(
+        compute='_compute_start_request_date',
+        string='Start Date of maintenance request completion',
+        store=True)
+    start_request_datetime_required = fields.Boolean(
+        compute='_compute_start_request_datetime_required')
+    end_request_datetime = fields.Datetime(
+        string='End Date and time of maintenance request completion')
+    end_request_date = fields.Date(
+        compute='_compute_end_request_date',
+        string='End Date of maintenance request completion',
+        store=True)
+    end_request_datetime_required = fields.Boolean(
+        compute='_compute_end_request_datetime_required')
+    real_request_duration = fields.Float(
+        compute='_compute_real_request_duration',
+        string='Real Duration',
+        store=True)
+    duration_difference = fields.Float(
+        compute='_compute_duration_difference',
+        string='Difference Duration',
+        store=True)
+    completion_delay = fields.Float(
+        compute='_compute_completion_delay',
+        string='Completion Delay',
+        store=True)
+
+    @api.depends('start_request_datetime')
+    def _compute_start_request_date(self):
+        for req in self:
+            req.start_request_date = \
+                (req.start_request_datetime.date()
+                 if req.start_request_datetime else None)
+
+    def _compute_start_request_datetime_required(self):
+        for req in self:
+            req.start_request_datetime_required = \
+                req.stage_id.done or req.stage_id.in_progress
+
+    @api.depends('end_request_datetime')
+    def _compute_end_request_date(self):
+        for req in self:
+            req.end_request_date = \
+                (req.end_request_datetime.date()
+                 if req.end_request_datetime else None)
```

Comment:
Alternative and clearer code:

```suggestion
            req.end_request_date = \ 
                req.end_request_datetime and req.end_request_datetime.date()
```

Reply:
Pending change (only suggestion)

Reply:
ok. The same on line 56...